src/main/java/org/openmrs/addonindex/scheduled/FetchDetailsToIndex.java

```diff
@@ -147,7 +148,8 @@ void fetchExtraDetailsForEachVersion(AddOnToIndex toIndex, AddOnInfoAndVersions
 					if (configXml == null) {
 						throw new IllegalArgumentException("No config.xml file in " + version.getDownloadUri());
 					} else {
-						handleConfigXml(configXml, version);
+						handleConfigXml(configXml, version, infoAndVersions, idAndPackageSet);
+						idAndPackageSet = true;
```

Comment:
Ideally we want to make sure that we do this only for the most recent version.
If this iteration is already sorted, then what you've done here is fine, but just add a comment saying that this is already sorted with the latest version first so we just do this the first time.